File: Data Structures & Algorithms/eating-bananas/submission-0.js
class Solution {
    /**
     * @param {number[]} piles
     * @param {number} h
     * @return {number}
     */
    minEatingSpeed(piles, h) {
        let l = 1
        let r = Math.max(...piles)
        let res = r

        while (l <= r){
            const mid = Math.trunc((l+r)/2)
            let times = 0
            for (const pile of piles){
                times += Math.ceil(pile/mid)
            }
            if (times <= h){
                res = mid
                r = mid-1
            } else  {
                
                l = mid+1
            }
            
        }
        return res
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-0.go
func trap(height []int) int {
    if len(height) == 0 {return 0} 
    res := 0
    l, r  := 0, len(height) - 1
    mL, mR := height[l], height[r]

    for l < r {
        if mL < mR {
            l++ 
            mL = max(mL, height[l])
            res += mL - height[l]
        } else { 
            r--
            mR = max(mR, height[r])
            res += mR - height[r]
        }
    }

    return res
}   

func max(a, b int) int {
    if a > b {return a} 
    return b
}